### DISCUSSION TITLE
Docker Plots

### DISCUSSION THREAD

**Discussion by kaitj** [OpenBB-finance/OpenBB#227] [General]
Mentioned in the README file is:
> Note: The problem with docker is that it won't output matplotlib figures.

In the past, I've built containers with python3-tk for `matplotlib`. Tested it out locally on Ubuntu 20.04 and am able to output the figures. Happy to make a PR (later tonight) if it would help, but wasn't sure how this problem was originally going to be tackled. While it makes it possible to output figures, the display still needs to be connected. I've included a screenshot below as an example (note the `-e DISPLAY=$DISPLAY` and `-v /tmp/.X11-unix:/tmp/.X11-unix`

![image](https://user-images.githubusercontent.com/21226986/111661428-65327c00-87e5-11eb-9173-d39d216f2bf9.png)

**Comment by aia** [comment]
Sounds great!

**Comment by hinxx** [comment]
Not sure, if this was merged or not, but I can confirm that it plays nicely with Linux based container on Linux host, as OP reports:
```
docker run -it --env="DISPLAY" --volume="/tmp/.X11-unix:/tmp/.X11-unix:rw" --rm --name GST gst
```
I wonder it Linux based container would be able to output plots when ran on OSX/Windoze host.

**Reply by hinxx** [reply]
> I wonder it Linux based container would be able to output plots when ran on OSX/Windoze host.

I followed this after installing the docker (and some WSL stuff) on windows:

https://dev.to/darksmile92/run-gui-app-in-linux-docker-container-on-windows-host-4kde

Then I imported (or was it load?) the docker image I created (and saved) in Linux directly from windows terminal:
```
c:\> docker run-it --rm -e DISPLAY=172.18.33.81:0.0 gst
```
The IP address was obtained with the `ipconfig`.

This shows that it would be possible to get the graphics out of GST Linux docker image, while running it under windoze. This could be an enabler to get the folks that do not wan to fiddle with the python installation and packages to quickly have GST up and running. BTW, I do not have access to OSX myself to see how that would work there..

**Reply by kaitj** [reply]
> Not sure, if this was merged or not, but I can confirm that it plays nicely with Linux based container on Linux host, as OP reports:

The inclusion of `python3-tk` was merged a while back (in #231). I haven't had time to look at more recent changes to the terminal or make another PR to update with instructions.

> This shows that it would be possible to get the graphics out of GST Linux docker image, while running it under windoze. 

Fantastic - glad to hear that it works there too. I would imagine it runs similarly in OSX with the intel chips - unsure about the new chips (though this is more of a Docker concern than a plot concern). Like you I do not have access to test. :man_shrugging:

**Reply by DidierRLopes** [reply]
Hey @kaitj and @hinxx , this sounds great! I have a OSX, I can try it out. 

But I don't have much experience docker, so you're gonna have to give me some instructions on how to test this!!

**Reply by kaitj** [reply]
I believe it should be similar to the commands used on a Linux host and work with either `docker` or `docker-compose` (unless something has changed since I last tested). Only part I am not sure about is the display on OSX - can do a bit of digging this weekend to see what I can find.

**Comment by ciekawy** [comment]
any chance to get plot showing up on docker/macos?

**Reply by DidierRLopes** [reply]
Did you try this https://github.com/OpenBB-finance/OpenBBTerminal/blob/main/DOCKER_ADVANCED.md? It should work!
